src/libstrongswan/plugins/openssl/openssl_kem.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2024 Amazon.com Inc. or its affiliates.
+ *
+ * Copyright (C) secunet Security Networks AG
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.  See <http://www.fsf.org/copyleft/gpl.txt>.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * for more details.
+ */
+
+/**
+ * @defgroup openssl_kem openssl_kem
+ * @{ @ingroup openssl_p
+ */
+
+#ifndef OPENSSL_KEM_H_
+#define OPENSSL_KEM_H_
+
+typedef struct openssl_kem_t openssl_kem_t;
+
+#include <crypto/key_exchange.h>
+
+/**
+ * Implementation of KEM algorithms using OpenSSL.
+ */
+struct openssl_kem_t {
```

Comment:
This indirection is not really necessary. `openssl_kem_create()` can just return a `key_exchange_t*` (see e.g. the example in `wolfssl_kem.h` or `botan_kem.h`). Also see the comments regarding the implementation.